src/transformers/models/wav2vec2/feature_extraction_wav2vec2.py

```diff
@@ -181,9 +181,12 @@ def __call__(
                 "Failing to do so can result in silent errors that might be hard to debug."
             )
 
-        is_batched = bool(
-            isinstance(raw_speech, (list, tuple))
-            and (isinstance(raw_speech[0], np.ndarray) or isinstance(raw_speech[0], (tuple, list)))
+        is_batched_numpy = isinstance(raw_speech, np.ndarray) and len(raw_speech.shape) > 1
+        if is_batched_numpy:
+            if len(raw_speech.shape) > 2:
+                raise ValueError(f"Only mono-channel audio is supported for input to {self}")
```

Comment:
I'm OK with throwing an error but something we might consider doing is mixing the audio to mono, which would be done by taking the mean over the channels. If I'm training a big model and one of my examples happens to be stereo, I'd rather that this automatically gets turned into mono (perhaps with a warning message) than fail.

Reply:
Probably you'd want this mixing to happen outside of the feature extraction code an in your data loading code no? IMO the purpose of the feature extraction should be kept as streamlined as possible in mapping a mono input to input features / input values, extra data pre-processing / checks can happen in the data loading steps

Reply:
The way I see it, the feature extractor could easily take care of stuff like resampling and conversion to mono etc. But this is a design decision: if you say that it's up to the user to provide the feature extractor with data in exactly the correct input format, then that's OK with me too. It just makes the feature extractor less convenient and kind of assumes you'll always be using this in combination with something like `datasets` or have to write your own additional resampling / conversion code.

Reply:
I think it's more logical to put in the data loader, but I wouldn't mind adding a bit to the docs here so it doesn't surprise anyone who's about to load a big dataset and crash things halfway through when a non-mono appears. How does sound?